### DISCUSSION TITLE
6.4.2 프론트엔드 배포

### DISCUSSION THREAD

**Discussion by wondongkyu** [fsoftwareengineer/todo-application#98] [질문과 답변]
실습코드 6-17과 같이 eb create --elb-type application --instance-type t2.micro 28s 를 파워쉘로 프로젝트 디렉토리 밑으로 생성 하려고 했지만 사진처럼 오류가 나네요
![캡처](https://user-images.githubusercontent.com/84184691/158290351-21233b34-89f5-4f95-926e-c41f09fb917a.PNG)

문제 해결을 위해서 뒤에 28s 를 지우고 만들어봤으나 그러면 aws 환경에서 502 에러가 발생하고 그걸 해결해보려고 다른분들이 해본것을 다해봤음에도 해결이 안되서 질문남깁니다

**Comment by fsoftwareengineer** [comment]
28s 뿐만 아니라 t2.micro이후의 공백도 지운 후 재시도 해 주시기 바랍니다. 
Enter Environment Name이 뜨지 않는 것으로 보아 28s나 공백이 EnvironmentName 으로 들어가는 것 같으니 둘 다 안된다면 'eb create ENVIRONMENTAL_NAME --elb-type...' 처엄 create다음에 환경 이름을 명시 해 주시기 바랍니다.

**Comment by fsoftwareengineer** [comment]
502애러와 cli에러는 서로 다른 문제로 인해 발생한 것으로 생각합니다.

**Reply by wondongkyu** [reply]
[BundleLogs-1647315774383.zip](https://github.com/fsoftwareengineer/todo-application/files/8249980/BundleLogs-1647315774383.zip)
cli 에러는 해결했습니다 그럼 aws 환경에서의 오류는 왜 나는것일까요 로그 요청해서 web.stdout 파일에서는 npm ERR! code ELIFECYCLE 라고나와서 node 모듈과 json 패키지 파일을 지웠다 재설치했는데도 오류가 남아있습니다 npm ERR! code ELIFECYCLE 라는 경고가 나오기 직전의 줄에는 /var/app/current/node_modules/.bin/react-scripts: Permission denied 이런 로그가 있었습니다

로그 파일도 남겨 드립니다

**Reply by fsoftwareengineer** [reply]
1. Node 버전 문제 일 수도 있다고 하니 node 버전을 변경 후 배포 하는 방법을 고려 해 보실 수 있습니다.
- https://github.com/typeorm/typeorm/issues/1122

2. node-cron` 모듈 때문에 에러가 날 수 있다고 하는데, 확인 해 주시기 바랍니다. 
- https://stackoverflow.com/questions/55022219/getting-npm-err-code-elifecycle-npm-err-errno-126-while-running-npm-install


해보고 안되면 또 다른 방법을 찾아보도록 하겠습니다. 화이팅!
